x-pack/plugins/security/server/audit/audit_service.ts

```diff
@@ -5,53 +5,181 @@
  */
 
 import { Subscription } from 'rxjs';
-import { Logger } from '../../../../../src/core/server';
-import { SecurityLicense } from '../../common/licensing';
+import { map, distinctUntilKeyChanged } from 'rxjs/operators';
+import {
+  Logger,
+  LoggingServiceSetup,
+  KibanaRequest,
+  HttpServiceSetup,
+  LoggerContextConfigInput,
+} from '../../../../../src/core/server';
+import { SecurityLicense, SecurityLicenseFeatures } from '../../common/licensing';
 import { ConfigType } from '../config';
+import { SpacesPluginSetup } from '../../../spaces/server';
+import { AuditEvent, httpRequestEvent } from './audit_events';
+import { SecurityPluginSetup } from '..';
 
-export interface AuditLogger {
+/**
+ * @deprecated
+ */
+export interface LegacyAuditLogger {
   log: (eventType: string, message: string, data?: Record<string, any>) => void;
 }
 
+export interface AuditLogger {
+  log: (event: AuditEvent) => void;
+}
+
+interface AuditLogMeta extends AuditEvent {
+  session?: {
+    id: string;
+  };
+  trace: {
+    id: string;
+  };
+}
+
 export interface AuditServiceSetup {
-  getLogger: (id?: string) => AuditLogger;
+  asScoped: (request: KibanaRequest) => AuditLogger;
+  getLogger: (id?: string) => LegacyAuditLogger;
 }
 
 interface AuditServiceSetupParams {
   license: SecurityLicense;
   config: ConfigType['audit'];
+  logging: Pick<LoggingServiceSetup, 'configure'>;
+  http: Pick<HttpServiceSetup, 'registerOnPostAuth'>;
+  getCurrentUser(
+    request: KibanaRequest
+  ): ReturnType<SecurityPluginSetup['authc']['getCurrentUser']> | undefined;
+  getSpaceId(
+    request: KibanaRequest
+  ): ReturnType<SpacesPluginSetup['spacesService']['getSpaceId']> | undefined;
 }
 
 export class AuditService {
+  /**
+   * @deprecated
+   */
   private licenseFeaturesSubscription?: Subscription;
-  private auditLoggingEnabled = false;
+  /**
+   * @deprecated
+   */
+  private allowAuditLogging = false;
+  private readonly betaLogger: Logger;
 
-  constructor(private readonly logger: Logger) {}
+  constructor(private readonly logger: Logger) {
+    this.betaLogger = this.logger.get('beta');
```

Comment:
**question** Can we use the same logger for both the legacy and new loggers now that this one isn't `beta`?